js/main.js

```diff
@@ -1,71 +1,93 @@
-    $parentRow.appendChild($columnHalf);
+    $parentContainer.appendChild($columnHalf);
   }
-});
-xhr.send();
+  data.allChampionData = xhr.response;
+}
 
 document.addEventListener('click', handleCardClick);
+
+var $allButtons = document.querySelectorAll('button');
 function handleCardClick(event) {
-  if (event.target.tagName === 'IMG') {
-    var xhr2 = new XMLHttpRequest();
-    var $parentCard = event.target.closest('div');
-    var $h2 = event.target.previousSibling;
-    var $img = $parentCard.querySelector('img');
-    if (!$parentCard.className.includes('lore')) {
-      xhr2.open('GET', 'http://ddragon.leagueoflegends.com/cdn/11.22.1/data/en_US/champion/' + $h2.textContent + '.json');
-      xhr2.responseType = 'json';
-      xhr2.addEventListener('load', function getChampionData() {
-        var individualChampionData = xhr2.response;
-        var $lore = document.createElement('p');
-        $lore.textContent = individualChampionData.data[$h2.textContent].lore;
-        $parentCard.appendChild($lore);
-        $parentCard.className += ' lore';
-        $img.className = 'dark';
-      });
-      xhr2.send();
-    } else if ($parentCard.className.includes('lore')) {
-      removeLore();
+  if (event.target.tagName === 'BUTTON') {
+    showRole(event.target.textContent);
+    for (var i = 0; i < $allButtons.length; i++) {
+      if (event.target.textContent === $allButtons[i].textContent) {
+        $allButtons[i].className = 'selected';
+      } else { $allButtons[i].className = ''; }
     }
-  } else if (event.target.tagName === 'P') {
-    removeLore();
+  } else if (event.target.tagName === 'IMG' || event.target.tagName === 'P') {
+    toggleLore();
   }
 }
 
-function removeLore() {
+var $allCards = document.querySelectorAll('.column-half');
+function showRole(role) {
+  for (var i = 0; i < $allCards.length; i++) {
+    if (!$allCards[i].getAttribute('roles').includes(role)) {
+      $allCards[i].style.display = 'none';
+    } else { $allCards[i].style.display = 'flex'; }
+  }
+}
+
+function toggleLore() {
   var $parentCard = event.target.closest('div');
   var $img = $parentCard.querySelector('img');
   var $lore = $parentCard.querySelector('p');
-  $parentCard.className = 'card';
-  $lore.remove();
-  $img.className = '';
+  if (!$parentCard.className.includes('lore')) {
+    $parentCard.className += ' lore';
+    $img.className = 'dark';
+    $lore.className = 'loreShown';
+  } else {
+    $parentCard.className = 'card';
+    $img.className = '';
+    $lore.className = '';
+  }
 }
 
 function capitalizeFirstLetter(string) {
   return string.charAt(0).toUpperCase() + string.slice(1);
 }
+
+// FUNCTION TO FLIP CARD LATER
+// function flipCard() {
+//   var $parentCard = event.target.closest('div');
+//   $parentCard.className = 'card flipped';
+//   hideLore();
+//   var $h3 = $parentCard.querySelector('h3');
+//   var $h2 = $parentCard.querySelector('h2');
+//   $h3.className = 'off';
+//   $h2.className = 'off';
+// }
```

Comment:
Please remove commented code prior to submitting for review. Remember, a feature only has content for that feature, nothing is included that isn't functional yet.